src/NUnitFramework/framework/Constraints/CollectionTally.cs

```diff
@@ -104,5 +87,92 @@ public void TryRemove(IEnumerable c)
             foreach (object o in c)
                 TryRemove(o);
         }
+
+        private class MissingItemCollection : IEnumerable<object>
```

Comment:
❔ Why introduce this class rather than keeping _items and _count as fields of CollectionTally?

Reply:
In introduced this class to minimise changes to the rest of the CollectionTally code.

Reply:
It feels like we're introducing a second collection tally class though. I'm not comfortable with it.